lib/sdl/cocoa_sdl_helpers.mm

```diff
@@ -55,3 +56,57 @@ bool cocoaIsSDLWindowFullscreened(SDL_Window *window)
 	assert(info.info.cocoa.window != nil);
 	return cocoaIsNSWindowFullscreened(info.info.cocoa.window);
 }
+
+std::string cocoaGetCurrentExecutablePath()
+{
+	std::string result;
+	char* path = (char*)malloc(MAXPATHLEN);
```

Comment:
can't `path` be allocated in the stack?

Reply:
I've tweaked this to allocate a base buffer of `MAXPATHLEN` on the stack, falling back to heap allocation only if `_NSGetExecutablePath` specifies it needs more.